src/types.rs

```diff
@@ -3,15 +3,83 @@
 
 use libc::{__s32, __u16, __u32, __u64, __u8, c_int};
 use std::ffi::OsStr;
+use std::os::fd::RawFd;
 
 // For documentaton linking
 #[allow(unused_imports)]
 use crate::api::*;
 #[allow(unused_imports)]
 use crate::flags::*;
 
+/// An inter convertable between [`RawFd`] struct that hold fd and provides
```

Comment:
you could've just used https://doc.rust-lang.org/stable/std/os/fd/struct.OwnedFd.html

Reply:
refactored in #16 fbce977c47cde0960a08b81aace5e0d48c73d5fe